prometheus/registry.go

```diff
@@ -722,5 +722,22 @@ func (s metricSorter) Less(i, j int) bool {
 			return vi < vj
 		}
 	}
-	return true
+
+	// While this is not strictly supported, until we have an
+	// import API some people are having some success getting
+	// data imported by supplying the same metric with different
+	// timestamps.  At the very least we shouldn't make things
+	// more difficult, so ensure that metrics get ordered by
+	// timestamp so that Prometheus doesn't complain about
+	// out-of-order metrics. A missing timestamp (implies "now")
+	// will be ordered last.
```

Comment:
Sorry for being picky, but I really don't want to suggest in any way here that you could do bulk import in that way. (Even your current success may cease to work with any new commit we do.) Suggestion for the comment:

"We should never arrive here. Multiple metrics with the same label set in the same scrape will lead to undefined ingestion behavior. However, as above, we have to provide stable sorting here, even for inconsistent metrics. So sort equal metrics by their timestamp, with missing timestamps (implying "now") coming last."